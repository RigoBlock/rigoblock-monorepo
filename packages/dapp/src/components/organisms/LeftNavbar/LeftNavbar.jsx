import './LeftNavbar.scss'
import AccountView from '../AccountView'
import NavigationView from '../NavigationView'
import React from 'react'

const LeftNavbar = () => {
  return (
    <div className="left-navbar">
      <AccountView />
      <div className="navbar-divider" />
      <NavigationView />
    </div>
  )
}

export default LeftNavbar
