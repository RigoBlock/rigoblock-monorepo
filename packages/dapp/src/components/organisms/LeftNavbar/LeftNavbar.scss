@import "../../_settings/variables.scss";

.left-navbar {
  padding-left: $dist-m;
  min-width: 20 * $dist-xs;

  &:first-child {
    margin-top: $dist-2xl;
  }
  > *:not(:first-child) {
    margin-top: $dist-m;
  }

  .left-navbar-divider {
    width: $dist-l;
    height: 1px;
    margin: $dist-xl 0;
    background-color: $dark-grey;
  }
}
