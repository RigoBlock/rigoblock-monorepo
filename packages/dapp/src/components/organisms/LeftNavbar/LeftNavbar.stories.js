import '../../_settings/_base.scss'
import * as ROUTES from '../../../constants/routes'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { selectV2, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import LeftNavbar from './LeftNavbar'
import React from 'react'

const mockStore = {
  getState: () => ({
    blockChain: {
      accounts: {
        '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196': {
          provider: 'metamask'
        },
        '0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d': {
          provider: 'metamask'
        },
        '0x8bb7481495d45ccd5cffae1c3a84155fea85a323': {
          provider: 'metamask'
        },
        '0xe198d98b76c529886affb5a74be8b435624bd310': {
          provider: 'metamask'
        },
        '0x927aa991a628aab6851a890fb790a4a7cd0ec446': {
          provider: 'metamask'
        }
      }
    },
    preferences: {
      currentAccount: '0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196'
    },
    routing: {
      location: {
        pathname: selectV2('url', ROUTES, ROUTES.DASHBOARD)
      }
    }
  }),
  dispatch: () => null,
  subscribe: () => null
}

storiesOf('Organisms/LeftNavbar', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .addDecorator(story => <Provider store={mockStore}>{story()}</Provider>)
  .add('default', () => <LeftNavbar />)
