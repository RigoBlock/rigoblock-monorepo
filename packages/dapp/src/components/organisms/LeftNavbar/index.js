import LeftNavbar from './LeftNavbar'

export default LeftNavbar
