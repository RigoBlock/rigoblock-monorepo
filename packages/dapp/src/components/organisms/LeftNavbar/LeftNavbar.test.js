import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LeftNavbar from './LeftNavbar.jsx'

describe('LeftNavbar component', () => {
  it('renders correctly', () => {
    expect(toJson(shallow(<LeftNavbar />))).toMatchSnapshot()
  })
})
