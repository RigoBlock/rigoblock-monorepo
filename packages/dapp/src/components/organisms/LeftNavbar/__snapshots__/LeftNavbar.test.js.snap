// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNavbar component renders correctly 1`] = `
<div
  className="left-navbar"
>
  <WrapperWithDivider
    Divider={[Function]}
  >
    <Connect(AccountView) />
    <Connect(NavigationView) />
  </WrapperWithDivider>
</div>
`;
